packages/nextjs/test/config/mocks.ts

```diff
@@ -0,0 +1,56 @@
+// TODO: This mocking is why we have to use `--runInBand` when we run tests, since there's only a single temp directory
+// created
+
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as rimraf from 'rimraf';
+
+import { CLIENT_SDK_CONFIG_FILE, SERVER_SDK_CONFIG_FILE } from './fixtures';
+
+// We use `fs.existsSync()` in `getUserConfigFile()`. When we're not testing `getUserConfigFile()` specifically, all we
+// need is for it to give us any valid answer, so make it always find what it's looking for. Since this is a core node
+// built-in, though, which jest itself uses, otherwise let it do the normal thing. Storing the real version of the
+// function also lets us restore the original when we do want to test `getUserConfigFile()`.
+export const realExistsSync = jest.requireActual('fs').existsSync;
+export const mockExistsSync = (path: fs.PathLike): ReturnType<typeof realExistsSync> => {
+  if ((path as string).endsWith(SERVER_SDK_CONFIG_FILE) || (path as string).endsWith(CLIENT_SDK_CONFIG_FILE)) {
+    return true;
+  }
+
+  return realExistsSync(path);
+};
+export const exitsSync = jest.spyOn(fs, 'existsSync').mockImplementation(mockExistsSync);
+
+/** Mocking of temporary directory creation (so that we have a place to stick files (like `sentry.client.config.js`) in
+ * order to test that we can find them) */
+
+// Make it so that all temporary folders, either created directly by tests or by the code they're testing, will go into
+// one spot that we know about, which we can then clean up when we're done
+const realTmpdir = jest.requireActual('os').tmpdir;
+
+// Including the random number ensures that even if multiple test files using these mocks are running at once, they have
+// separate temporary folders
+const TEMP_DIR_PATH = path.join(realTmpdir(), `sentry-nextjs-test-${Math.random()}`);
+
+jest.spyOn(os, 'tmpdir').mockReturnValue(TEMP_DIR_PATH);
+// In theory, we should always land in the `else` here, but this saves the cases where the prior run got interrupted and
+// the `afterAll` below didn't happen.
+if (fs.existsSync(TEMP_DIR_PATH)) {
+  rimraf.sync(path.join(TEMP_DIR_PATH, '*'));
+} else {
+  fs.mkdirSync(TEMP_DIR_PATH);
+}
+
+afterAll(() => {
+  rimraf.sync(TEMP_DIR_PATH);
+});
```

Comment:
Wait, if I register an `(before|after)(Each|All)` call outside a `describe` closure, does that mean it runs for all tests in this directory? 
If yes, that's totally fine, I just haven't seen this before, which is why I'm asking

Reply:
Nope, it only runs for the tests which import the file containing it. Neat, huh?

Honestly, before yesterday I didn't know that, either. I knew you could have a `(before|after)(Each|All)` call outside a `describe` in the same file (so, module level), but not that it could be in a totally separate file. But I took a flier and it ended up working out! (I tested it by logging inside the `afterAll()` function, and it only logged three times, once from inside of each file importing `mocks.ts`.)

Reply:
Interesting, but makes sense! Thanks for explaining!